src/win/rlimit.c

```diff
@@ -87,17 +82,15 @@ int getrlimit(int resource, struct rlimit *rlp)
   //
   // Check to make sure the resource value is within range
   //
-  if((resource < 0) || (resource >= RLIM_NLIMITS))
-  {
-    iRet = EINVAL;
-  }
+  if(resource < 0 || resource >= RLIM_NLIMITS)
+    return EINVAL;
 
   //
   // Return both rlim_cur and rlim_max
   //
   *rlp = rlimits[resource];
```

Comment:
Code checks if `resource` is a valid index, but accesses the array `rlimits` not matter if `resource` not out-of-bound or not.